### PR TITLE
feat: support product server

### PR DESCRIPTION
**Pull request by zllkjc**
# PR Details

split dev server and product server, to reduce the size

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/server/product-server/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Your Package
```
Comment (chenjiahan) [inline packages/server/product-server/README.md:1]: 标题忘改咯

File: packages/server/product-server/package.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
+{
+  "name": "@modern-js/product-server",
```
Comment (chenjiahan) [inline packages/server/product-server/package.json:2]: product 是 production 的意思吗？标准的缩写应该是 `prod`，product 会有歧义

File: packages/solutions/app-tools/src/commands/start.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { logger, chalk } from '@modern-js/utils';
 import { useAppContext, useResolvedConfigContext } from '@modern-js/core';
-import server from '@modern-js/server';
+import server from '@modern-js/product-server';
```
Comment (chenjiahan) [inline packages/solutions/app-tools/src/commands/start.ts:3]: 推荐用 named exports/imports

File: packages/server/product-server/README.md (hunk 1 above)
Comment (zllkjc) [reply packages/server/product-server/README.md]: done

File: packages/solutions/app-tools/src/commands/start.ts (hunk 3 above)
Comment (zllkjc) [reply packages/solutions/app-tools/src/commands/start.ts]: 这里是有默认导出也有 named exports 的。并不会用 server.xxx。

**Review by yimingjfe — APPROVED** [review]

**Pull request merged**